src/java.base/share/classes/sun/util/locale/LanguageTag.java

```diff
@@ -414,13 +419,8 @@ private boolean parsePrivateuse(StringTokenIterator itr, ParseStatus sts) {
     }
 
     public static String caseFoldTag(String tag) {
-        ParseStatus sts = new ParseStatus();
-        parse(tag, sts);
-        // Illegal tags
-        if (sts.errorMsg != null) {
-            throw new IllformedLocaleException(String.format("Ill formed tag:" +
-                    " %s", sts.errorMsg));
-        }
```

Comment:
If we want to keep the existing behavior, catch the ILE and throw a new one with the prepending text, but I think that won't be necessary.

Reply:
Yes, I didn't think it was needed to do that, the existing message should already have enough details. I filed the CSR just in case, I'll withdraw it.